### PR TITLE
Support extension on decorated number

### PR DESCRIPTION
**Pull request by mouseyeung**

### CODE REVIEW

File: src/decorators/gb.ts [hunk 1]
```diff
@@ -81,9 +83,19 @@ const decorator: IDecorator = {
...
     }
 
+    let isExtension = false;
+
     for (let i = 0; i < phoneNumber.length; ++i) {
       const digit = phoneNumber[i];
-      decoratedNumber.parts.push(numberPart(digit));
+
+      if (digit === "x") {
+        decoratedNumber.parts.push(extensionDecorativePart(" ext. "));
```
Comment (ChrisH91) [inline src/decorators/gb.ts]: Part of this issue with the extensions stuff is that there isn't a well defined convention for how to display it. `ext.` seems to be used be some people, some places, but it only works in English.

German numbers do have a well defined format for displaying numbers.

I think the most sensible option is passing in the delimiter when you instantiate the library.

File: src/decorators/nanp.ts [hunk 2]
```diff
@@ -37,6 +41,11 @@ const decorator: IDecorator = {
 
       if (i >= 0 && i <= 2) {
         decoratedNumber.parts.push(areaCodePart(digit));
+      } else if (digit === "x") {
```
Comment (ChrisH91) [inline src/decorators/nanp.ts]: Again, this isn't really a known convention, and I'm not even sure it's a sensible default outside of English language countries. I think passing this in on instantiation as well is most sensible.

**Review by ChrisH91 — CHANGES_REQUESTED** [review]

File: src/decorators/au.ts [hunk 3]
```diff
@@ -60,6 +63,15 @@ const decorator: IDecorator = {
         }
       }
 
+      if (digit === "x") {
+        decoratedNumber.parts.push(extensionDecorativePart(" ext. "));
```
Comment (JPeaker) [inline src/decorators/au.ts]: I think we need baked in support for localization here. Obviously on the LoopUp side, but also the purity of the library as a whole shouldn't have us depending on hardcoded strings

File: src/decorators/gb.ts (hunk 1 above)
Comment (JPeaker) [reply src/decorators/gb.ts]: My comment above is also in line with what Chris says here. I think the delimiter and replacement should be provided on instantiation

**Review by JPeaker — CHANGES_REQUESTED** [review]

File: src/decorators/gb.ts (hunk 1 above)
Comment (brgmnn) [reply src/decorators/gb.ts]: Ok, I think that makes sense as well.

**Review by ChrisH91 — APPROVED** [review]

**Pull request merged**